### ISSUE TITLE
[FEA]: Implement Series.to_frame()

### ISSUE THREAD

**Issue by TomAugspurger** [rapidsai/cudf#1161]
```
In [13]: pd.Series.to_frame?
Signature: pd.Series.to_frame(self, name=None)
Docstring:
Convert Series to DataFrame.

Parameters
----------
name : object, default None
    The passed name should substitute for the series name (if it has
    one).

Returns
-------
data_frame : DataFrame

In [14]: s = pd.Series([1, 2])

In [15]: s.to_frame()
Out[15]:
   0
0  1
1  2

In [16]: s.to_frame(name='name')
Out[16]:
   name
0     1
1     2
```

For now, the following works

```
In [27]: cudf.DataFrame({0 if s.name is None else s.name: s})
Out[27]: <cudf.DataFrame ncols=1 nrows=2 >
```

**Comment by quasiben** [comment]
I think this is already in the [cudf branch-0.6](https://github.com/rapidsai/cudf/blob/branch-0.6/python/cudf/dataframe/series.py#L179-L182) -- done in PR https://github.com/rapidsai/cudf/pull/985

**Comment by TomAugspurger** [comment]
Indeed, thanks.

Looks like https://github.com/rapidsai/cudf/pull/985/files#diff-573959c9fbfbe8e637119d34065be555R181 has a small bug for falsey names (if those are allowed in cudf):

```
In [2]: s = pd.Series([1, 2], name=False)

In [3]: s.to_frame()
Out[3]:
   False
0      1
1      2
```

Looks like that would return a DataFrame  with the columns `[0]` (haven't checked on master).

**Comment by quasiben** [comment]
@TomAugspurger https://github.com/rapidsai/cudf/pull/1168

**Comment by kkraus14** [comment]
Fixed by #1168
